### PR TITLE
Clean up projection inside storage snapshot

### PR DESCRIPTION
**Pull request by amosbird**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
It aims to complete #58630. This is made possible by #60463, #61459 and #60082.

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Comment by robot-clickhouse-ci-1** [comment]
*This is an automated comment for commit e48b862b368f950b8321a773e120a72650470051 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/66443/e48b862b368f950b8321a773e120a72650470051/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/ad21a9307fc1d0715388923d8f1e490a7bd9bebe/docs_check.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

**Comment by alesapin** [comment]
cc @davenger

**Comment by davenger** [comment]
> cc @davenger

Looks Ok to me

File: src/Storages/StorageSnapshot.h [hunk 1]
```diff
@@ -82,11 +82,6 @@ struct StorageSnapshot
     void check(const Names & column_names) const;
 
     DataTypePtr getConcreteType(const String & column_name) const;
-
```
Comment (CurtizJ) [inline src/Storages/StorageSnapshot.h:85]: Please also remove the field `projection`.

**Review by CurtizJ — APPROVED** [review]
LGTM

**Review by CurtizJ — APPROVED** [review]

**Pull request merged**